dali/pipeline/executor/async_separated_pipelined_executor.h

```diff
@@ -44,11 +44,16 @@ class DLL_PUBLIC AsyncSeparatedPipelinedExecutor : public SeparatedPipelinedExec
         gpu_thread_(device_id, set_affinity) {}
 
   DLL_PUBLIC ~AsyncSeparatedPipelinedExecutor() override {
+    Shutdown();
+  }
+
+  DLL_PUBLIC void Shutdown() override {
     ShutdownQueue();
 
     cpu_thread_.ForceStop();
     mixed_thread_.ForceStop();
     gpu_thread_.ForceStop();
+    SyncDevice();
```

Comment:
```suggestion
    SeparatedPipelinedExecutor::Shutdown();
```

Reply:
Mhhh, won't `~AsyncSeparatedPipelinedExecutor()` call `Executor<WorkspacePolicy, QueuePolicy>::~Executor()` and sync anyway?

Reply:
It will be too late - some stuff will get destroyed which we want to be kept around.

Reply:
Understood.